core/domain/takeout_service_test.py

```diff
@@ -1007,14 +1236,16 @@ def test_export_data_nontrivial(self):
             'platform_parameter_snapshot_metadata':
                 expected_platform_parameter_sm,
         }
+
         user_takeout_object = takeout_service.export_data_for_user(
             self.USER_ID_1)
         observed_data = user_takeout_object.user_data
         observed_images = user_takeout_object.user_images
-        self.assertEqual(observed_data, expected_data)
+        self.assertItemsEqual(observed_data, expected_user_data)
         observed_json = json.dumps(observed_data)
-        expected_json = json.dumps(expected_data)
-        self.assertEqual(json.loads(observed_json), json.loads(expected_json))
+        expected_json = json.dumps(expected_user_data)
+        self.assertItemsEqual(
+            json.loads(observed_json), json.loads(expected_json))
```

Comment:
Perhaps check, say, that the user ID string can be found within the observed_json but the profile's user ID string is not present.

(And that a subscription -- or some other item -- belonging to the profile user ID can't be found, but the corresponding item can be found for the full user ID.)

Reply:
Regarding your suggestion: In takeout since we never export the `user_id` string, the only place I could find `user_id` string in `observed_json` was - `'general_feedback_email_reply_to_id': expected_reply_to_data` where the expected string "user_1_reply_to_id_thread_1" has been hardcoded for the test, so I think it doesn't make sense searching for profile id string naively.

My solution: For `UserSkillMasteryModel` and `CompletedActivitiesModel` I assigned different exploration ids and collections ids to ensure data for full and profile user for corresponding models is absolutely exhaustive (if by mistake profile's entry for any model gets exported, the test would fail). And similarly for full user in profile user tests, the data for models which are populated for both full user and profile user - data has been made different. 

Sounds good?

Reply:
I have made a separate test now. Now I am verifying that None of the populated models for profiles are present in the takeout data.